src/javascript/app/hooks/events.js

```diff
@@ -0,0 +1,84 @@
+// events.js
+import { useEffect, useState } from 'react';
+
+class EventEmitter {
+    constructor() {
+        this.events = {};
+    }
+
+    subscribe(event, callback) {
+        if (!this.events[event]) {
+            this.events[event] = [];
+        }
+        this.events[event].push(callback);
+        return () => {
+            this.events[event] = this.events[event].filter((cb) => cb !== callback);
+        };
+    }
+
+    emit(event, data) {
+        if (this.events[event]) {
+            this.events[event].forEach((callback) => callback(data));
+        }
+    }
+}
+
+const eventEmitter = new EventEmitter();
+
+// Custom hooks
+const useCustomEvent = (eventName) => {
+    const [hasNewChanges, setHasNewChanges] = useState(false);
+
+    useEffect(() => {
+        const handleEvent = () => {
+            setHasNewChanges(true);
+            setTimeout(() => setHasNewChanges(false), 0);
+        };
+
+        const unsubscribe = eventEmitter.subscribe(eventName, handleEvent);
+
+        return () => {
+            unsubscribe();
+        };
+    }, [eventName]);
+
+    return hasNewChanges;
+};
+
+export const eventDispatcher = (element, eventName) => {
+    const event = new Event(eventName, { bubbles: true, cancelable: true });
+    element.dispatchEvent(event);
+};
+
+// Trigger functions
+const triggerCustomEvent = (eventName) => {
+    eventEmitter.emit(eventName);
```

Comment:
I wonder if this is it the only way of handling global state in the application? As far as I can see, all the below events are related to state management. Can we use a context or a store instead of emitting custom events globally and keeping state in the global window object?

Reply:
Im not sure context will work in the legacy js that we have. We want to keep it simple and only update the UI and not refactor the current script flow.